src/parse_instruction.py

```diff
@@ -278,15 +296,13 @@ def constant_fold(arg: Argument) -> Argument:
     return arg
 
 
-def replace_bare_reg(arg: Argument, arch: ArchAsmParsing) -> Argument:
+def replace_bare_reg(arg: Argument, arch: ArchAsmParsing, used_reg_names: UsedRegNames) -> Argument:
     """If `arg` is an AsmGlobalSymbol whose name matches a known or aliased register,
     convert it into a Register and return it. Otherwise, return the original `arg`."""
     if isinstance(arg, AsmGlobalSymbol):
         maybe_reg = Register(arg.symbol_name)
-        if maybe_reg in arch.aliased_regs:
-            return arch.aliased_regs[maybe_reg]
         if maybe_reg in arch.all_regs:
-            return maybe_reg
+            return used_reg_names.input_to_internal(maybe_reg, arch)
```

Comment:
aliased_regs aren't included in all_regs, so this needs to check `in all_regs or in aliased_regs`

Reply:
I'm surprised it works at all if that's the case.